openjdk/mmtkBarrierSetAssembler_x86.cpp

```diff
@@ -36,13 +36,32 @@
 void MMTkBarrierSetAssembler::eden_allocate(MacroAssembler* masm, Register thread, Register obj, Register var_size_in_bytes, int con_size_in_bytes, Register t1, Label& slow_case) {
   assert(obj == rax, "obj must be in rax, for cmpxchg");
   assert_different_registers(obj, var_size_in_bytes, t1);
-  // printf("eden_allocate\n");
+
   if (!MMTK_ENABLE_ALLOCATION_FASTPATH) {
     __ jmp(slow_case);
   } else {
-    // We always use the default allocator.
-    // But we need to figure out which allocator we are using by querying MMTk.
-    AllocatorSelector selector = get_allocator_mapping(AllocatorDefault);
+    // MMTk size check. If the alloc size is larger than the allowed max size for non los,
+    // we jump to slow path and allodate with LOS in slowpath.
+    // Note that OpenJDK has a slow path check. Search for layout_helper_needs_slow_path and FastAllocateSizeLimit.
+    // I tried to set FastAllocateSizeLimit in MMTkHeap::initialize(). But there are still large objects allocated into the
+    // default space.
+    size_t max_non_los_bytes = get_max_non_los_default_alloc_bytes();
```

Comment:
If we had a constant, instead of this function call, would that affect performance?

Reply:
I measured it, and there was no measurable difference. I think among all these compiler code, this is just one call (calling to Rust has no extra overhead). So it doesn't matter much. And it is not actually a constant though, it depends on the selected plan.

Reply:
This code will only be executed by the assembler when it compiles some Java code. So it will become a compile-time inlined constant in the generated machine code.

Reply:
This will be fired at each Java code compilation. Is it possible to cache `max_non_los_bytes` into a global variable? so we only need to call `get_max_non_los_default_alloc_bytes` once at the beginning. We are not switching plans at runtime, so `get_max_non_los_default_alloc_bytes`  should always return the same value.

Reply:
But I feel that we can't really measure the difference. So having the call inside assembler is also fine.

Reply:
I created a static variable in `MMTkMutatorContext` to cache the value, the value is initialized right after we call `gc_init()`. When we read the value, there are assertions to make sure the value is initialized. But I don't think it matters for performance.